string/test/src/io/github/iltotore/iron/test/string/CompileTimeSpec.scala

```diff
@@ -0,0 +1,55 @@
+package io.github.iltotore.iron.test.string
+
+import io.github.iltotore.iron.*, constraint.{*, given}, string.constraint.{*, given}
+import io.github.iltotore.iron.test.UnitSpec
+
+class CompileTimeSpec extends UnitSpec {
+
+  "A LowerCase constraint" should "return Right if the argument is lower case" in {
+
+    def dummy(x: String / LowerCase): String / LowerCase = x
+
+    "dummy(\"abc\")" should compile 
+    "dummy(\"ABC\")" shouldNot compile
+  }
+
+  "An UpperCase constraint" should "return Right if the argument is upper case" in {
+
+    def dummy(x: String / UpperCase): String / UpperCase = x
+
+    "dummy(\"ABC\")" should compile 
+    "dummy(\"abc\")" shouldNot compile 
+  }
+
+  "A custom Match[V] without MatchConstraint" should "always compile" in {
+
+    def dummy(x: String / Match["^[a-z0-9]+"]): String / Match["^[a-z0-9]+"] = x
+
+    "dummy(\"abc123\")" should compile
+    "dummy(\"abc\")" should compile
+    "dummy(\"123\")" should compile
+    "dummy(\" \")" should compile
+    "dummy(\"$!#\")" should compile
+  }
+
+  "A custom Match[V] with MatchConstraint" should "detect regex at compile time" in {
+
+    type alphanumeric = Match["^[a-z0-9]+"]
```

Comment:
Your type alias doesn't respect Scala naming conventions. Is it intentional ?

Reply:
I've done it to separate if from `Alphanumeric ` from constraint.scala - but I'll rename to to `TestAlphanumeric` or I'll create brand new regex.

Reply:
You can still use the `Alphanumeric` you created before. Since it's a "transparent" type alias, it is assumed it is the same as `Match["^[a-z0-9]+"]`

Reply:
I've created new Regex so it won't accidentally mix with the existing ones.